modules/c++/sys/include/sys/Conf.h

```diff
@@ -223,14 +224,52 @@ namespace sys
             {
                 innerOff = offset + j;
                 innerSwap = offset + elemSize - 1 - j;
-                
+
                 std::swap(bufferPtr[innerOff], bufferPtr[innerSwap]);
             }
         }
     }
 
     /*!
-     *  Function to swap one element irrespective of size.  The inplace 
+     *  Swap bytes into output buffer.  Note that a complex pixel
+     *  is equivalent to two floats so elemSize and numElems
+     *  must be adjusted accordingly.
+     *
+     *  \param buffer to transform
+     *  \param elemSize
+     *  \param numElems
+     *  \param[out] outputBuffer buffer to write swapped elements to
+     */
+    inline void  byteSwap(const void* buffer,
+                          unsigned short elemSize,
+                          size_t numElems,
+                          void* outputBuffer)
+    {
+        const sys::byte* bufferPtr = static_cast<const sys::byte*>(buffer);
+        sys::byte* outputBufferPtr = static_cast<sys::byte*>(outputBuffer);
+
+        if (!numElems || !bufferPtr || !outputBufferPtr)
+        {
+            return;
+        }
+
+        unsigned short half = elemSize >> 1;
+        size_t offset = 0, innerOff = 0, innerSwap = 0;
+
+        for (size_t ii = 0; ii < numElems; ++ii, offset += elemSize)
+        {
+            for (unsigned short jj = 0; jj < half; ++jj)
+            {
+                innerOff = offset + jj;
+                innerSwap = offset + elemSize - 1 - jj;
+                std::memcpy(&outputBufferPtr[innerOff], &bufferPtr[innerSwap], 1);
```

Comment:
Since you're just copying one byte, and these things are cast to be `sys::byte` pointers anyway, I think this would be easier to follow if it was just an assignment:

```
outputBufferPtr[innerOff] = bufferPtr[innerSwap];
outputBufferPtr[innerSwap] = bufferPtr[innerOff];
```